### ISSUE TITLE
Multiple private git repo dependencies got error

### ISSUE THREAD

**Issue by niheaven** [r-lib/remotes#597]
When I develop some package that depends on multiple other packages that live on private repo, I got some error that `remotes::install_deps()` only install last one and omit others.

I use the following to add deps:

```r
usethis::use_dev_package("a", remote = "git::http://private.repo/git/a.git")
usethis::use_dev_package("b", remote = "git::http://private.repo/git/b.git")
```

And the `DESCRIPTION` likes:

```
Imports:
	a,
	b
Remotes:
	git::http://private.repo/git/a.git,
	git::http://private.repo/git/b.git
```

But when I try to `install_deps` and actually install the developed package, R only install dep package `b`, twice 😭 

```r
install_deps()
# Skipping 2 packages not available: a, b
# Downloading git repo http://private.repo/git/b.git
# xxxxxxxxxxx
# Downloading git repo http://private.repo/git/b.git
# xxxxxxxxxxx
```

What's wrong? Or is it a bug?

**Comment by niheaven** [comment]
When there is only one private repo, `install_deps` works well. So maybe syntax is correcr, I thought.

I try to figure out related code that may cause this error, but it jumps in and out too many packages and C code. So I really need developer's help😓

**Comment by michael-dewar** [comment]
I had either the same or a similar error with multiple private repos.  Whenever I tried to install my package A, regardless of how I tried (`install()`, `install_github()`, `install_local()`) I kept getting an error about lacking credentials to install package B, which was also in a private repo.  This only happened on one of my computers (my deployment server) and not the other (my development laptop).  I imagine there is a problem with getting the GitHub credentials for installing the dependencies.

I overcame the problem by moving all private packages from Depends to Suggests in the DESCRIPTION file.
